coder_sniffer/Drupal/Sniffs/Commenting/TodoCommentSniff.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Parses and verifies comment language.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+namespace Drupal\Sniffs\Commenting;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+/**
+ * Parses and verifies that comments use the correct @todo format.
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+class TodoCommentSniff implements Sniff
+{
+
+
+    /**
+     * Returns an array of tokens this test wants to listen for.
+     *
+     * @return array<int|string>
+     */
+    public function register()
+    {
+        return [
+            T_COMMENT,
+            T_DOC_COMMENT_TAG,
+        ];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+        // Use the comment text by default.
+        $comment = $tokens[$stackPtr]['content'];
+        // Use the next line for multi-line comments.
+        if ($tokens[$stackPtr]['code'] === T_DOC_COMMENT_TAG) {
+            $comment = $phpcsFile->findNext(T_DOC_COMMENT_STRING, ($stackPtr + 1));
+        }
+
+        $expression = '/^\s*(?i)(?=(@*to(-|\s|)+do))(?-i)(?!@todo\s{1}[A-Z])/m';
+        if ((bool) preg_match($expression, $comment) === true) {
+            $phpcsFile->addError('@todo comments should be in the "@todo Some task." format.', $stackPtr, 'TodoFormat');
```

Comment:
You could have
````
$comment = trim($comment, " /\r\n");
$phpcsFile->addError("'%s' should match the format '@todo Some task'", $stackPtr, 'TodoFormat', [$comment]);
````
Then the user sees what is wrong. This also requires that the tag text is appended to the original $comment, so line 56 would be
````
$tagText  = $phpcsFile->findNext(T_DOC_COMMENT_STRING, ($stackPtr + 1));
$comment .= ' '.$tokens[$tagText]['content'];
````

Reply:
Actually, we need to do the appending above regardless of whether we use the $comment substitution in the message. It is needed because the `@whatever` tag will be in the original $comment variable, and this is needed in the final text that is used in the preg_match. I think it is reasonable to just append with a space between, as this will not affect the matching outcome.

Reply:
Figuring this all out took some thinking, as I'm new to writing code sniffs, but I think I sorted it out. Thanks for the guidance here.